### PR TITLE
Allow yarn 2 resolutions

### PR DESCRIPTION
**Pull request by mhassan1**
- [X] Tests written and linted
- [ ] Documentation written / README.md updated [https://snyk.io/docs/snyk-for-node/](i)
- [X] Follows [CONTRIBUTING agreement](CONTRIBUTING.md)
- [X] Commit history is tidy [https://git-scm.com/book/en/v2/Git-Branching-Rebasing](i)
- [ ] Reviewed by Snyk team

### What this does

Yarn 2 `resolutions` breaks `snyk test`. See https://github.com/snyk/nodejs-lockfile-parser/issues/99 for details. This PR unblocks consumers that have yarn 2 resolutions when `--strict-out-of-sync=false` is passed to the Snyk CLI.

### Notes for the reviewer

See additional test case for `yarn 1` and yarn 2` resolutions.

### CODE REVIEW

File: test/lib/fixtures/resolutions/yarn1/expected-tree.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "dependencies": {
+    "adm-zip": {
+      "labels": {
+        "scope": "prod"
+      },
+      "name": "adm-zip",
+      "version": "0.4.7"
```
Comment (mhassan1) [inline test/lib/fixtures/resolutions/yarn1/expected-tree.json:8]: This is actually not correct. The actual version of `adm-zip` that has been installed is `0.4.8`, due to `resolutions`. I'm not attempting to fix it in this PR.

File: test/lib/fixtures/resolutions/yarn2/expected-tree.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "dependencies": {
+    "adm-zip": {
+      "labels": {
+        "scope": "prod",
+        "missingLockFileEntry": "true"
+      },
+      "name": "adm-zip",
+      "version": "0.4.7"
```
Comment (mhassan1) [inline test/lib/fixtures/resolutions/yarn2/expected-tree.json:9]: This is actually not correct. The actual version of `adm-zip` that has been installed is `0.4.8`, due to `resolutions`. I'm not attempting to fix it in this PR.

File: test/lib/fixtures/resolutions/yarn2/yarn.lock [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  resolution: "adm-zip@npm:0.4.8"
+  checksum: f08c1f94d892545327ed925ea38861cb401ff1f2b496123eb9cc0d1be57546fb0337fda3b77b75c41b8a59d165aa38d2a97a5dc5a825fc0471da3c3bb858e238
+  languageName: node
+  linkType: hard
+
+"goof@workspace:.":
+  version: 0.0.0-use.local
+  resolution: "goof@workspace:."
+  dependencies:
+    adm-zip: 0.4.7
```
Comment (mhassan1) [inline test/lib/fixtures/resolutions/yarn2/yarn.lock:18]: The dependency is on `0.4.7`, which does not actually appear elsewhere in the `yarn.lock`. This is expected behavior for yarn 2.

File: test/lib/fixtures/resolutions/yarn1/yarn.lock [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+adm-zip@0.4.7:
```
Comment (mhassan1) [inline test/lib/fixtures/resolutions/yarn1/yarn.lock:5]: Yarn 1 includes this dependency in the `yarn.lock`, even though it is not installed. This is expected behavior for yarn 1.

**Comment by orsagie** [comment]
Hi @mhassan1 
Thanks for your contribution 🙏 . I have added a task for us to go over the PR this sprint.

**Comment by JamesPatrickGill** [comment]
Hey @mhassan1 
We have been looking into getting this contribution merged and have merged these commits into another PR (https://github.com/snyk/nodejs-lockfile-parser/pull/102) to squash the commits and get the checks to pass. Interestingly once the checks passed on the PR we they passed over here too.

We are going to merge your commits into master from the other PR and close this as that is our current process. 

Thanks for your contribution 😀

**Pull request closed without merging**